### PR TITLE
auction: define action balances

### PR DESCRIPTION
**Pull request by erwanor**
## Describe your changes

This PR:
- add `auctionnft`s to the asset registry
- augment `AuctionNft` with denom metadata
- implements `DutchAuctionDescription::id`
- implements value `balance` for each action
- implements computing the balance commitment for the withdraw action

## Issue ticket number and link
This is a prelude to #4212 

## Checklist before requesting a review

- [x] If this code contains consensus-breaking changes, I have added the "consensus-breaking" label. Otherwise, I declare my belief that there are not consensus-breaking changes, for the following reason:

  > This isn't connected to the rest of the app yet.

### CODE REVIEW

File: crates/core/component/auction/src/auction/dutch/actions/withdraw.rs [hunk 1]
```diff
@@ -15,6 +17,44 @@ pub struct ActionDutchAuctionWithdraw {
...
+    ///
+    /// More context: [Actions and Value balance][protocol-spec]
+    /// [protocol-spec]: https://protocol.penumbra.zone/main/transactions.html#actions-and-value-balance
+    pub fn balance_commitment(&self) -> balance::Commitment {
+        let prev_auction_nft = Balance::from(Value {
+            amount: 1u128.into(),
+            // The sequence number should always be >= 1, because we can
+            // only withdraw an auction that has ended (i.e. with sequence number `>=1`).
+            // We use a saturating operation defensively so that we don't underflow.
+            asset_id: AuctionNft::new(self.auction_id, self.seq.saturating_sub(1)).asset_id(),
```
Comment (erwanor) [inline crates/core/component/auction/src/auction/dutch/actions/withdraw.rs:44]: My rationale to do this is that we check the binding signature before we run any action validation checks, so even if this MUST be `seq >= 1`, we can't actually rely on that yet.

File: crates/core/component/auction/src/auction/nft.rs [hunk 2]
```diff
@@ -1,15 +1,31 @@
...
+    pub seq: u64,
+    /// The metadata corresponding to the nft denom.
+    pub metadata: asset::Metadata,
+}
+
+impl AuctionNft {
+    pub fn new(id: AuctionId, seq: u64) -> AuctionNft {
+        let metadata = asset::REGISTRY
+            .parse_denom(&format!("auctionnft_{seq}_{id}"))
+            .expect("auction nft denom is valid");
```
Comment (erwanor) [inline crates/core/component/auction/src/auction/nft.rs:22]: I think I want to do a second pass on both auction and lp nfts to make their constructor fallible.

File: crates/core/component/auction/src/auction/dutch.rs [hunk 3]
```diff
@@ -65,6 +69,28 @@ pub struct DutchAuctionDescription {
     pub nonce: [u8; 32],
 }
 
+impl DutchAuctionDescription {
+    /// Compute the unique identifier for the auction description.
+    pub fn id(&self) -> AuctionId {
+        let mut state = blake2b_simd::Params::default()
+            .personal(DUTCH_AUCTION_DOMAIN_SEP)
```
Comment (redshiftzero) [inline crates/core/component/auction/src/auction/dutch.rs:76]: I think the personalization bytes should be maximum 16 bytes: https://docs.rs/blake2b_simd/latest/blake2b_simd/struct.Params.html#method.personal

**Review by redshiftzero — CHANGES_REQUESTED** [review]
Value balance changes LGTM! one small comment inline

File: crates/core/component/auction/src/auction/dutch.rs (hunk 3 above)
Comment (erwanor) [reply crates/core/component/auction/src/auction/dutch.rs:76]: oh great catch

**Review by redshiftzero — APPROVED** [review]

**Pull request merged**